GoogleApiWrappers/src/Google.Apis.Storage.v1.ClientWrapper/StorageClient.DownloadObject.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright 2015 Google Inc. All Rights Reserved.
+// Licensed under the Apache License Version 2.0.
+
+using Google.Apis.Download;
+using Google.Common;
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Object = Google.Apis.Storage.v1.Data.Object;
+
+namespace Google.Apis.Storage.v1.ClientWrapper
+{
+    public partial class StorageClient
+    {
+        // Note: documentation will be provided when we've worked out if we really want 16 methods here.
+        // See issue #21.
+
+        public void DownloadObject(string bucket, string sourceName, Stream destination)
```

Comment:
The underlying API refers to the GCS object's identity as simply `object`. The API docs define the `object` parameter as "Name of the [GCS] object":
https://cloud.google.com/storage/docs/json_api/v1/objects/get

Since `object` won't worry for obvious reasons, and I agree with you that `path` has its own connotations, how about `gcsObject` here and later in the Upload case (instead of `destination`.)

I really don't care how we dress up the term "object", but I do think we should use "object" to match the underlying APIs and documentation for GCS rather than using a more vague term like `sourceName`.

Though, I'm happy to admit this isn't the end of the world. From the usage context it's pretty clear what each of these parameters are for, regardless of how they are named.